src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Uno.Extensions;
+using Uno.UI.RemoteControl.HotReload.Messages;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Markup;
+
+namespace Uno.UI.RemoteControl.HotReload
+{
+	public class ClientHotReloadProcessor : IRemoteControlProcessor
+	{
+		public const string Scope = "hotreload";
+
+		private string _projectPath;
+		private string[] _xamlPaths;
+
+		private readonly IRemoteControlClient _rcClient;
+
+		public ClientHotReloadProcessor(IRemoteControlClient rcClient)
+		{
+			_rcClient = rcClient;
+		}
+
+		string IRemoteControlProcessor.Scope => Scope;
+
+		public async Task Initialize()
+		{
+			await ConfigureServer();
+		}
+
+		public async Task ProcessFrame(Messages.Frame frame)
+		{
+			switch (frame.Name)
+			{
+				case FileReload.Name:
+					ReloadFile(JsonConvert.DeserializeObject<HotReload.Messages.FileReload>(frame.Content));
+					break;
+			}
+		}
+
+		private async Task ConfigureServer()
+		{
+			if (_rcClient.AppType.Assembly.GetCustomAttributes(typeof(ProjectConfigurationAttribute), false) is ProjectConfigurationAttribute[] configs)
+			{
+				var config = configs.First();
+
+				_projectPath = config.ProjectPath;
+				_xamlPaths = config.XamlPaths;
+
+				if (this.Log().IsEnabled(LogLevel.Debug))
+				{
+					this.Log().LogDebug($"ProjectConfigurationAttribute={config.ProjectPath}, Paths={_xamlPaths.Length}");
+				}
+			}
+			else
+			{
+				if (this.Log().IsEnabled(LogLevel.Error))
+				{
+					this.Log().LogError("Unable to find ProjectConfigurationAttribute");
+				}
+			}
+
+			await _rcClient.SendMessage(new HotReload.Messages.ConfigureServer(_projectPath, _xamlPaths));
+		}
+
+		private async Task ReloadFile(FileReload fileReload)
+		{
+			Windows.UI.Core.CoreDispatcher.Main.RunAsync(
+				Windows.UI.Core.CoreDispatcherPriority.Normal,
+				async () =>
+            {
+                try
+                {
+                    if (this.Log().IsEnabled(LogLevel.Debug))
+                    {
+                        this.Log().LogDebug($"Reloading changed file [{fileReload.FilePath}]");
+                    }
+
+                    var uri = new Uri("file:///" + fileReload.FilePath.Replace("\\", "/"));
+
+                    foreach (var instance in EnumerateInstances(Window.Current.Content, uri))
+                    {
+                        switch (instance)
+                        {
+#if __IOS__
```

Comment:
Why the iOS-only UserControl special case?

Reply:
Because of this legacy property:

https://github.com/unoplatform/uno/blob/38d3c0ded22601d79b5a1a006b63b37915912e0a/src/Uno.UI/UI/Xaml/Controls/UserControl/UserControl.iOSmacOS.cs#L29

We could probably change that to detect in the XamlReader and use the property underneath. It's a breaking change that we'll need to make at some point.